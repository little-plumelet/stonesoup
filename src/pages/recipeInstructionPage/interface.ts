export interface IIngredient {
  name: string;
  amount: number;
  unit: string;
}

export interface IDetailedRecipe {
  extendedIngredients: IIngredient[];
  id: number;
  title: string;
  aggregateLikes: number;
  image: string;
  readyInMinutes: number;
}
