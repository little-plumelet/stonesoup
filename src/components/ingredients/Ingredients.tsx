import { List, Typography } from 'antd';
import { IIngredient } from '../../interfaces/ingredient.interface';

interface IIngridentsProps {
  extendedIngredients: IIngredient[];
}

const { Title } = Typography;

export function Ingredients({ extendedIngredients }: IIngridentsProps) {
  return (
    <>
      <Title level={3}>Ingredients:</Title>
      <List
        size="small"
        dataSource={extendedIngredients}
        renderItem={(item: IIngredient) => (
          <List.Item>
            <List.Item.Meta description={item.name} />
            <div>
              {item.amount} {item.unit}
            </div>
          </List.Item>
        )}
      />
    </>
  );
}
