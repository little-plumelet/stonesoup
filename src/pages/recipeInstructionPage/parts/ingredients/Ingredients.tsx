import { List, Typography } from 'antd';
import { IIngredient } from '../../interface';

interface IIngridentsProps {
  extendedIngredients: IIngredient[];
}

const { Title } = Typography;

export function Ingredients({ extendedIngredients }: IIngridentsProps) {
  return (
    <>
      <Title level={3}>Ingridients:</Title>
      <List
        size="small"
        dataSource={extendedIngredients}
        renderItem={(item: IIngredient) => (
          <List.Item>
            <List.Item.Meta description={item.name} />
            <div>
              {item.amount} {item.unit}
            </div>
          </List.Item>
        )}
      />
    </>
  );
}
