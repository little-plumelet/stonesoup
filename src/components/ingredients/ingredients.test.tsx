import { render } from '@testing-library/react';
import { Ingredients } from './Ingredients';
import { IIngredient } from '../../interfaces/ingredient.interface';
import '../../../matchMedia';

const mockIngredients: IIngredient[] = [
  {
    name: 'garlic',
    amount: 2,
    unit: '',
  },
  {
    name: 'tomatillos',
    amount: 5.5,
    unit: 'pounds',
  },
];

describe('ingredients', () => {
  it('should render', () => {
    const component = render(
      <Ingredients extendedIngredients={mockIngredients} />
    );

    expect(component).toMatchSnapshot();
  });
});
