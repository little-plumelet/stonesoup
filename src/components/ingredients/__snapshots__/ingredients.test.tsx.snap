// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ingredients should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="ant-typography css-dev-only-do-not-override-mxhywb"
      >
        Ingredients:
      </h3>
      <div
        class="ant-list ant-list-sm ant-list-split css-dev-only-do-not-override-mxhywb"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-mxhywb"
        >
          <div
            class="ant-spin-container"
          >
            <ul
              class="ant-list-items"
            >
              <li
                class="ant-list-item"
              >
                <div
                  class="ant-list-item-meta"
                >
                  <div
                    class="ant-list-item-meta-content"
                  >
                    <div
                      class="ant-list-item-meta-description"
                    >
                      garlic
                    </div>
                  </div>
                </div>
                <div>
                  2
                   
                </div>
              </li>
              <li
                class="ant-list-item"
              >
                <div
                  class="ant-list-item-meta"
                >
                  <div
                    class="ant-list-item-meta-content"
                  >
                    <div
                      class="ant-list-item-meta-description"
                    >
                      tomatillos
                    </div>
                  </div>
                </div>
                <div>
                  5.5
                   
                  pounds
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h3
      class="ant-typography css-dev-only-do-not-override-mxhywb"
    >
      Ingredients:
    </h3>
    <div
      class="ant-list ant-list-sm ant-list-split css-dev-only-do-not-override-mxhywb"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-mxhywb"
      >
        <div
          class="ant-spin-container"
        >
          <ul
            class="ant-list-items"
          >
            <li
              class="ant-list-item"
            >
              <div
                class="ant-list-item-meta"
              >
                <div
                  class="ant-list-item-meta-content"
                >
                  <div
                    class="ant-list-item-meta-description"
                  >
                    garlic
                  </div>
                </div>
              </div>
              <div>
                2
                 
              </div>
            </li>
            <li
              class="ant-list-item"
            >
              <div
                class="ant-list-item-meta"
              >
                <div
                  class="ant-list-item-meta-content"
                >
                  <div
                    class="ant-list-item-meta-description"
                  >
                    tomatillos
                  </div>
                </div>
              </div>
              <div>
                5.5
                 
                pounds
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
